### PR TITLE
Sync tracker pool with headers pool

### PR DESCRIPTION
**Pull request by SebastianMarian**
* Sync tracker pool with headers pool LRU cacher when missing headers from tracker are requested (in this situation a delete will be done in headers pool, just to be sure that the notifiy event will be called for the requested/received header, even if in some edge case situations the requested header exists only in headers pool but not in tracker pool)

This situation could appear only in some rare events of restarting the node after more than 1000 rounds from its last commit

### CODE REVIEW

**Review by sasurobert — DISMISSED** [review]

File: process/block/baseProcess.go [hunk 1]
```diff
@@ -255,7 +255,9 @@ func (bp *baseProcessor) requestHeadersIfMissing(
 		missingNonces = append(missingNonces, nonces...)
 	}
 
+	headersPool := bp.dataPool.Headers()
```
Comment (iulianpascalau) [inline process/block/baseProcess.go:258]: some tests for the added code would be nice.

Comment (SebastianMarian) [reply process/block/baseProcess.go:258]: would be added until the system tests are done

Comment (iulianpascalau) [reply process/block/baseProcess.go:258]: great

**Review by sasurobert — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]
System tests passed.

**Pull request merged**